guides/components.md

```diff
@@ -113,6 +114,30 @@ andThen(function() {
 page.modal.clickOn("I'm sure");
 ```
 
+## Functions and Getters
+
+You can use native syntax for functions and getters on pages as well to create
+methods and properties with custom functionality:
+
+```js
+import { click } from 'ember-test-helpers';
+import { findElement } from 'ember-cli-page-object';
+
+const page = create({
+  button: {
+    scope: 'button',
+
+    clickWith(options) {
+      click(findElement(this), options);
```

Comment:
I think we can't recommend direct usage of external test helpers for actions. In order to implement a new type of action we require a private import `import { getExecutionContext } from 'ember-cli-page-object/-private/execution_context';` currently.

I believe we would [address](https://github.com/san650/ember-cli-page-object/commit/71adf959a35319196449f8cbea42bfb731c987ac) it soon. But for now we only support "read" functions and usages of actions defined on the page object context, like:
```
async fillIn(data) {
  await this.field1.fillIn('...');
  return await this.field2.fillIn('...');
}
```

Reply:
What about?

```
login(data) {
  return this.username(data.username)
    .password('secret', data.password)
    .submit();
}
```

Reply:
I don't think execution context has to be imported. 

If the user is using native context, they _know_ they can just use `ember-test-helpers` directly. If the user is using the old jquery context, they know they should be using jquery. We can address this in the guides, but using plain functions should be fine, we use them all the time in our tests.

The key here is that when using function syntax, you are _inherently_ making one off functions that are _not_ macros. For instance, see [these helpers in Ember Table](https://github.com/Addepar/ember-table/blob/master/addon-test-support/pages/ember-table.js#L62-L75);

Reply:
Actually I think I misread that. Agree that we can change the example to the one you suggested, since it doesn't have the same issue. I still think it's fine for users to use `findElement` as long as they know what execution context their app will be running in, but we would need to document that.

Reply:
Yeah, using of findElement is totally ok